### ISSUE TITLE
1.3.0.0 broke many things

### ISSUE THREAD

**Issue by felixbrucker** [nicehash/NiceHashMiner-Archived#36]
created everything from scratch with 1.3.0.0, noticed some bugs, glitches, whatever:
- lyra2rev2 benchmark generates invalid shares and outputs incorrect hashrate (0.6mh/s instead of 16mh/s)
- upon starting the main program an error is displayed: "ethminer.exe stopped working", even if no automining is set, just plain initializing of the program
- x11 hashrate is sometimes correct, but most times off a bit (precise test)
- auto profit switching does not work, currently eth is most profitable, while the miner uses x11, using advanced settings ("skip" is set to true as default now, dont know why) i was able to force eth by selecting skip=false for eth only
- eth mining crashes from time to time (<5min, ethminer.exe stopped working), whereas with plain ethminer.exe mining on nicehash (which i did before) it does not crash
- ethminer generated the dag every time i ran it, which is not normal

**Comment by theLosers106** [comment]
What Windows and GPU(s) are you using? Does Ethminer runs OK if you run it with the command line?

**AMD**:

> `ethminer --opencl --opencl-platform X -F http://eth-mine.suprnova.cc:3000/suprnova.5/10 --dag-dir ..\..\data`

*Change the X to whatever platform number your GPU(s) are on.

**nVidia**: 

> `ethminer --cuda -F http://eth-mine.suprnova.cc:3000/suprnova.5/10 --dag-dir ..\..\data`

**Comment by d57heinz** [comment]
similar results here aswell on lyra2re.. same happening with hd 7970 win 10 pro 16 gb ram with crimson 16.2 newest drivers.  Last version works just fine.  many of the benchmark timed out but could be that their is no mining available for those coins.. .. Ethereum did work on the benchmark for me.. Havent tried to mine with it yet.. Figured id let you get the bugs worked out first before i bring my eth farm over.   Just a thought is their a way you guys can setup a benchmark server so that we dont have to have the need for a coin to be available at the time fo the benchmark.Maybe a testnet for the coins??. seems it would be nice if their was a feature like ethminer -G - M but for the sgminers.>  Thanks for the continued work on this miner. Going to be that much more awesome once we get the bugs out of it..Going to do some more testing now ill report back asap  Best Regards
Doug

**Comment by theLosers106** [comment]
@d57heinz thanks (: .. could you try replacing the sgminer-5-3-0-general with this one and see if the benchmark problem is fixed:
[https://drive.google.com/open?id=0B6kB9L8GNvQwN3lTZDlFZFVaME0](url)

**Comment by mendoza1468** [comment]
With 2XGtx970 Eth bench is at 36 mh/s but after 10 min of mining he is slower than 10 mh/s, auto profit switching does not work. Stay on Eth with low hashrate. I had to uncheck eth

**Comment by theLosers106** [comment]
@mendoza1468 what Windows are you using? That's weird why your cards slow down after awhile of mining.. what speed does it say on the ethminer itself?

**Comment by mendoza1468** [comment]
Système d'exploitation
Windows 10 ‎(X64)‎
Date d'installation: 17/11/2015
Processeur
Intel Core i7 6700K Cadencé à 4 GHz
Fréquence mesurée (core0): 4 GHz
Nombre de coeurs: 4 physique(s), 8 logique(s)
Socket: Socket 1151 LGA
Température CPU (core 0): 32 °C
Carte mère
Gigabyte Technology Co., Ltd. Z170X-Gaming 3 To be filled by O.E.M.
Numéro de série (carte mère): To be filled by O.E.M.
SKU Number: To be filled by O.E.M.
Version du bios: F6
Date: 01/03/2016
Chipset
Pont nord : Intel Skylake
Identifiant de révision: 07
Pont sud : Intel Skylake PCH
Identifiant de révision: 31
Mémoire
16 Go de mémoire totale de type DDR4 à 1.60 GHz
Timings mémoire: 16.0 clocks-18 clocks-18 clocks-38 clocks-2 TT
Barrette G.Skill de 8 Go
Fréquence maximale: 1.07 GHz
Bande passante: DDR4-2133
Barrette G.Skill de 8 Go
Fréquence maximale: 1.07 GHz
Bande passante: DDR4-2133
Carte Graphique
NVIDIA GeForce GTX 970
Version DirectX installée: 11.0
NVIDIA GeForce GTX 970
Version DirectX installée: 11.0
Disque dur
SSD Crucial_CT250MX200SSD1 de 232.89 Go SATA III

**Comment by mendoza1468** [comment]
![1](https://cloud.githubusercontent.com/assets/16471209/13561867/d3f9284e-e3ff-11e5-990f-74d97b13adca.JPG)
![2](https://cloud.githubusercontent.com/assets/16471209/13561868/d3f99766-e3ff-11e5-97d5-5d3ca587cec9.JPG)

![3](https://cloud.githubusercontent.com/assets/16471209/13561913/58242056-e400-11e5-8ed3-84f56d6f0987.JPG)

![4](https://cloud.githubusercontent.com/assets/16471209/13561985/1ae60a82-e401-11e5-8885-bc4aa43588f7.JPG)

2 minute after look at the hash speed (2 pictures). Never hit the 36 mh/s or not even near :( 
Other algo seems to be ok for now.

But you guys doin a good job. Thanks !!

**Comment by felixbrucker** [comment]
my machines are:
- win 7 x64 sp1, 1x amd 7870, 15.7
- win 7 x64 sp1, 2x amd 7970, 15.12
- win 8.1 x64, 1x amd 290x, 16.1

im working today, i can test the eth miner in the evening, the commands i was using to manually run ethminer were:
- ethminer.exe -G -F http://ethereum.eu.nicehash.com:3500/n1c3-1KoYpyVLBeThpbihUwDghno1S2SWvSDyRX.rig2/14 --opencl-device 0 --opencl-platform 0
- ethminer.exe -G -F http://ethereum.eu.nicehash.com:3500/n1c3-1KoYpyVLBeThpbihUwDghno1S2SWvSDyRX.rig1/42
- ethminer.exe -G -F http://ethereum.eu.nicehash.com:3500/n1c3-1KoYpyVLBeThpbihUwDghno1S2SWvSDyRX.home/32 --opencl-device 0 --opencl-platform 1

(the order is the machine order from above)

**Comment by theLosers106** [comment]
@mendoza1468 does this happen if you run the miner normally using the command line? thanks (:

**Comment by theLosers106** [comment]
@felixbrucker I think I may know what is causing the crashes for ethminer and possibly other miners.. could you right click on the ethminer.exe and click on the properties.. make sure it is unblocked.. if not, it will crash the miner..

![image](https://cloud.githubusercontent.com/assets/15833762/13562372/fcf5c068-e472-11e5-8fa2-31b63f3523d3.png)

**Comment by WhyteDem0n** [comment]
I'm having issues with ethminer crashing as well. I'm on Win10 with a GTX970. I checked the file properties as @theLosers106 mentioned and it's not blocked.

**Comment by mendoza1468** [comment]
I use the software, no command line.

**Comment by felixbrucker** [comment]
@theLosers106 

> ethminer.exe -G -F http://ethereum.eu.nicehash.com:3500/n1c3-1KoYpyVLBeThpbihUwDghno1S2SWvSDyRX.home/32 --opencl --opencl-platform 1 --dag-dir ..\..\data

created a new dag in the data folder, which structure is now:
- full-R23-4b77c50b31c33143 (1.3GB)
- full-R23-eada66563356d409 (300MB) NEW
- full-R23-5197b2a3898b31d4 (1.3GB) NEW
- AMD_OpenCL -> full-R23-4b77c50b31c33143 (1.3GB)
- AMD_OpenCL -> full-R23-eada66563356d409 (1.3GB)

and it does not crash after successfully creating the dag

---

for me (win 8.1 x64) there is no such "unblock" thing, and im having UAC disabled

---

right now (i did nothing except starting ethminer manually for the test above) the Nicehashminer software starts **without** "ethminer.exe stopped working"

---

i tried starting ethereum mining via the software again: 
- created the dag (again) under AMD_OpenCL dir
- does not crash (at least in the early 3 minutes)

feeling confident i tried starting the miner with "window hidden" enabled, and it crashed with a similar/identical message as in #37 : 

> System.NullReferenceException: Der Objektverweis wurde nicht auf eine Objektinstanz festgelegt.
>    bei NiceHashMiner.Miner._Start()
>    bei NiceHashMiner.Miner.Restart()
>    bei NiceHashMiner.Form1.MinerStatsCheck_Tick(Object sender, EventArgs e)
>    bei System.Windows.Forms.Timer.OnTick(EventArgs e)
>    bei System.Windows.Forms.Timer.TimerNativeWindow.WndProc(Message& m)
>    bei System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

translation: Object reference not set to an object instance

---

in conclusion it seems as the ethereum mining crashes seem to be resolved by starting the miner via cli once, though i'm not sure how that should help :D 

---

another thing is the auto profit switching, i noticed all "Skip" bools are set to true, why is that?
i would love to use nicehashminer with eth while still being able to automatically switch to another crypto if needed, which currently does not seem to work

---

in regards of benchmark/lyra2rev2 sgminer problems i will just wait for a fixed binary (as invalid shares seem to be related to sgminer binary), though lyra2rev2 is coming right after eth for me

cheers

**Comment by WhyteDem0n** [comment]
Tried running via the command line once as per @felixbrucker suggested and let it create the DAG. After creating the dag, the client is able to open the eth miner successfully. However, it still crashes if I enable the window hiding.
![error](https://cloud.githubusercontent.com/assets/10343684/13587867/17279544-e499-11e5-9f23-e382eb6effbc.png)

**Comment by theLosers106** [comment]
To whomever that is running with HideMiningWindows enabled.. it will crash eth-proxy.. temporary fix would be to disable this.. it will be fixed in the next build..

If anyone would like to test the dev build of NiceHashMiner, here is the link:
https://drive.google.com/file/d/0B6kB9L8GNvQwZG0zeFN5dGF3UFE/view?usp=sharing

should fix the HideMiningWindows bug.. replace only the NiceHashMiner.exe

**Comment by mendoza1468** [comment]
Eth benchmark reach 33 mh/s precise with 2 gtx 970, but im stable at 15mh/s after 5 min, so its a huge difference for algo switch. Anyelse getting the same problem?
I tried to do a longer bench 240 sec but don't seem to change none.

Thanks

**Comment by felixbrucker** [comment]
those points are still open/not working:
- ethminer generates the dag every time i run it, which is not normal
- lyra2rev2 generates invalid shares

regarding the first point: upon starting mining one window with ethminer starts which generates the dag, sadly in the wrong dir (data), after that a new instance of ethminer and a second window with ether-proxy opens, this time ethminer uses the correct path (data/AMD_OpenCL) but there is no dag, so it starts creating a new one, which gets interrupted as the nicehashminer thinks the miner is dead and restarts it in the middle of the creation process.

the second point might be fixed by another bin, i guess it will be included in the next release

the "every algo is set to skip" might have been fixed with the patched exe, im not sure

cheers

**Comment by theLosers106** [comment]
@felixbrucker Hi.. it is intended to be that way as it is easier for people who have AMD + nVidia GPUs inside one rig.. generate once and copy to the AMD and nVidia folder.. for the lyra2rev2, try using a fresh config file.. it might be the settings are messed up and causing errors..

btw.. you could post your config.json file here too so I can see it..

**Comment by felixbrucker** [comment]
@theLosers106 well then its not working as expected as the copying does not seem to work/the miner tries to generate a new one and gets killed doing that

i initially started with a fresh 1.3.0.0 dir, no configs and nothing, or do you mean between original 1.3.0.0 exe and 1.3.0.1?

sure here is the config:
[config.json.zip](https://github.com/nicehash/NiceHashMiner/files/164940/config.json.zip)

**Comment by theLosers106** [comment]
@felixbrucker Could you make sure that you have enough space on the disk.. Can't see anything else that is causing the problem.. =/

**Comment by felixbrucker** [comment]
@theLosers106 there is 180GB free space on my SSD, can't be that

**Comment by theLosers106** [comment]
@felixbrucker hey.. can you test the new release and see if that fixed the problem..
https://github.com/nicehash/NiceHashMiner/releases/tag/1.3.0.1

**Comment by Heavyiron** [comment]
1.3.0.1 Problem with lyra2rev2 still present on radeon 7850 win 8.1 x64. Sgminer 5.1.0-optimized creates ckolivas bin.

**Comment by theLosers106** [comment]
@Heavyiron will be fixed in the next build.. sorry for the inconvenience.. please go to the config.json or Settings and change lyra2rev2 to Lyra2REv2.. make sure it has the same caps letters

**Comment by Heavyiron** [comment]
**theLosers106** Change lyra2rev2 to Lyra2REv2 does not work for me. I have tried to change this in the settings and manually in the config file. After restarting nicehash Lyra2REv2 again changed to lyra2rev2.

**Comment by theLosers106** [comment]
@Heavyiron try this.. should fix the problem.. https://drive.google.com/file/d/0B6kB9L8GNvQwcjQ1WHpyaGRSam8/view?usp=sharing

**Comment by Heavyiron** [comment]
**theLosers106** It works fine, thanks.

**Comment by theLosers106** [comment]
@Heavyiron will be in the next release.. @felixbrucker could you help me debug your problem so I could put it in the next release..

**Comment by felixbrucker** [comment]
@theLosers106 yes ethmining confirmed working with 1.3.0.1
will try the patched exe for lyra2rev2 and benchmark the whole system.

also the eth bench reports 41mh/s while the miner displays 32mh/s, ether-proxy reports the same as the miner
not a big deal, i can change the values in the advanced config dialog or the json file, just sayin

**Comment by felixbrucker** [comment]
i noticed another bug in regards to x11/qubit on one of my machines:

one of the gpus gets disabled because it "failed", other algos work fine

the rig in question has 1x 7970 and 1x 290x

**Comment by Balbec78** [comment]
GPU-fan  is out-of-control in 1.3.0.0 (1) on sgminer-5-3-0-general with my Juniper RADEON 57xx. Modifying configs in "extra-launch parameters" give no result (--gpu-fan 60-95 --auto-fan --auto-gpu) . Fan stays 30% load and no reaction on temperature raising over limits. sgminer-5-2-1-general in 1.2.3.2 works fine - about 60 C regulated. Seems ancients enjoy one another...

**Comment by malagent** [comment]
So upgraded to 1.3.0.1, all is wonderful for a while. Ethereum is mining well and I test it for a while to see what the real world results are. It doesn't seem to be anywhere near consistant in earnings, which I expect due to the immature and unconverted aspects. HOWEVER, it just suddenly stopped working entirely. When I launch the miners Ethminer will repeatedly startup and crash. When I click the Windows 10 Dialog box to close it, it restarts a few moments later and inantly or within a few seconds crashes again. I've had to disable Eth in order to use the NiceHash Miner at all, this is disappointing.

My equipment isnt's special but without messing with anything I got a day of working, then it just broke. Windows 10 64bit, Athlon 5350, AMD Radeon 57 250

![image](https://cloud.githubusercontent.com/assets/17844295/13893437/4d9a69da-ed2c-11e5-9ecd-180623d663cc.png)

**Comment by felixbrucker** [comment]
the bug regarding x11 seems to be version independant (tried 1.2.3.2) so i will open another issue for that, all my previous issues with 1.3.0.0 have been resolved except  for a little "annoyance":
when mining eth the first windows generates a dag in /data/, then copies it to /data/AMD_OpenCL and strats the real mining window which uses this copied file starts mining. By doing so it creates the new dag in /data/AMD_OpenCL which is missing in /data/, so when starting the miner software the first windows then creates the "new" dag from scratch while it exists in /data/AMD_OpenCL

**Comment by theLosers106** [comment]
@Balbec78 and @malagent .. please create a new issue as this one is relating to a different version and I'll be closing this one.. it makes it more organized that way.. thanks (:
